src/libraries/Microsoft.Extensions.Hosting.WindowsServices/src/WindowsServiceLifetimeHostBuilderExtensions.cs

```diff
@@ -106,11 +106,15 @@ public static IServiceCollection AddWindowsService(this IServiceCollection servi
 
         private static void AddWindowsServiceLifetime(IServiceCollection services, Action<WindowsServiceLifetimeOptions> configure)
         {
+#if !NETFRAMEWORK
             Debug.Assert(RuntimeInformation.IsOSPlatform(OSPlatform.Windows));
```

Comment:
These don't seem all that valuable.  According to https://github.com/dotnet/runtime/pull/61867/commits/04442bf3f869e1745fa0b668fb30c0b9bff007c5#r759693314 it's to workaround an issue in the analyzer.  I wonder if that issue still exists?

Reply:
> I wonder if that issue still exists?

Probably would still warn as I don't see any annotation or guard. Isn't it windows only? Then we might want to add `[SupportedOSPlatformGuard("windows")]` attribute to this type to suppress the warnings